### DISCUSSION TITLE
When using Notifications, app.status.operationState is nil

### DISCUSSION THREAD

**Discussion by jzaayman** [argoproj/argo-cd#7895] [Q&A]
I'm hoping someone can clear notifications up for me.

I've installed [notifications](https://argocd-notifications.readthedocs.io/en/stable/), and I subscribe to the `on-deployed` trigger in the `Application` with the following annotation:

```yaml
annotations:
  notifications.argoproj.io/subscribe.on-deployed.slack: my_channel
```

No notification is sent. If I modify `argocd-notifications-cm` and force the condition for `trigger.on-deployed` to `when: true` and put `{{ .app.status.operationState }}` in the output, a Slack message is sent, but the content is `<no-value>`. It appears that the `app.status.operationState` object is not set and so the original condition fails. But this is an Argo CD out-the-box.

Am I missing something?

Output of `argocd version` below:

```
argocd: v2.1.7+a408e29.dirty
  BuildDate: 2021-11-19T05:55:07Z
  GitCommit: a408e299ffa743213df3aa9135bf7945644ec936
  GitTreeState: dirty
  GoVersion: go1.17.2
  Compiler: gc
  Platform: darwin/amd64
argocd-server: v2.1.7+a408e29
  BuildDate: 2021-11-17T22:10:27Z
  GitCommit: a408e299ffa743213df3aa9135bf7945644ec936
  GitTreeState: clean
  GoVersion: go1.16.5
  Compiler: gc
  Platform: linux/amd64
  Ksonnet Version: v0.13.1
  Kustomize Version: v4.2.0 2021-06-30T22:49:26Z
  Helm Version: v3.6.0+g7f2df64
  Kubectl Version: v0.21.0
  Jsonnet Version: v0.17.0
```

**Comment by noisyscanner** [comment]
Did you ever find a solution to this?

**Comment by kintarowins** [comment]
still unanswered after all these years? I'm having the same problem with argocd 2.6
